File: design-patterns-2/ts/typescript/patterns/factories/abstractFactory/Runner.ts
import IRunner from "../../IRunner";
import Factory1 from "./Factory1";

export default class Runner implements IRunner{
    run(exampleNumber: number=1){
        const factory = new Factory1();

        const a = factory.createProductA();
        const b = factory.createProductB();

        console.log(a.operationA());
        console.log(b.operationB());
        console.log(b.combinedOperation(a));
    }
}

File: design-patterns-2/csharp/patterns/factories/abstractFactory/Runner.cs
namespace csharp.patterns.factories.abstractFactory;

public class Runner : IRunner
{
    public void Run(int exampleNumber)
    {
        var machine = new HotDrinkMachine();

        var myTea = machine.BrewDrink(HotDrinkMachine.DrinkToBrew.Tea, 12);
        myTea.Consume();

        var myCoffee = machine.BrewDrink(HotDrinkMachine.DrinkToBrew.Coffee, 3);
        myCoffee.Consume();
    }
}